auditable-serde/Cargo.toml

```
@@ -22,7 +22,7 @@ serde = { version = "1", features = ["serde_derive"] }
 serde_json = "1.0.57"
 semver = { version = "1.0", features = ["serde"] }
 cargo_metadata = { version = "0.15", optional = true }
-cargo-lock = { version = "8.0.2", default-features = false, optional = true }
+cargo-lock = { version = "9", default-features = false, optional = true }
```

An alternative that might work:

```suggestion
cargo-lock = { version = ">=8.0.2, <10", default-features = false, optional = true }
```

...though it would be a good idea to resolve with `-Z minimal-versions` and test against it if you go that route

---

I believe in this case Cargo will always choose v9 (without nightly flags that is) and there's no obvious way to override this, unlike with feature flags.

---

It will choose v8 if another dependency requires `^8`

---

Oh that's actually great. If you're sure it works that way, I would prefer to go this route then. I want to test it works that way though!

Aaand also I already merged :disappointed:

---

This approach *sort of* works. If you make a toplevel crate depending on `auditable-serde` that depends on `cargo-lock` 8, then everything works out. But if you switch to 9, compilation fails; but running `cargo update` fixes it again.

I feel this is confusing for API users, and I don't want to resort to this just yet.